src/AuthenticatedApp.js

```diff
@@ -1,175 +1,191 @@
-        }
-        return;
-      }
-
-      localStorage.setItem('__access_token', oauth.access_token);
-      if (this._isMounted) this.setState((state) => {
-        return {
-          ...state,
-          access_token: oauth.access_token
-        };
-      });
-
-      fetch('https://api.twitch.tv/helix/users', {
-        headers: {
-          'Client-ID': process.env.REACT_APP_TWITCH_CLIENT_ID,
-          Authorization: `Bearer ${oauth.access_token}`
-        }
-      })
-      .then(r => r.json())
-      .then(userInfo => {
-        //console.log(userInfo); //login [aka lowercase username?], display_name, profile_image_url, description
-        localStorage.setItem('__username', userInfo.data[0].login);
-        return fetch(`https://api.twitch.tv/helix/moderation/moderators?broadcaster_id=${userInfo.data[0].id}`, {
-          headers: {
-            'Client-ID': process.env.REACT_APP_TWITCH_CLIENT_ID,
-            Authorization: `Bearer ${this.state.access_token}`
-          }
-        })
-        .then(r => r.json())
-        .then(modInfo => {
-          const modList = (!modInfo.data) ? null : modInfo.data.map((modObj) => modObj.user_name.toLowerCase())
-          if (this._isMounted) {
-            this.setState((state) => {
-              return {
-                ...state,
-                username: userInfo.data[0].login,
-                modList
-              };
-            });
-          }
-        })
-      })
-    })
-  }
-
-  render() {
-    return (
-      <div>
-        {this.props.location.pathname === "/gamelist" && <JackboxGameList /> }
-        {this.state.failed_login && this.props.location.pathname !== "/gamelist"
-          ? <Redirect to="/login" />
-          : this.state.username && <MainScreen
-                                      channel={this.state.username}
-                                      modList={this.state.modList}
-                                      access_token={this.state.access_token}
-                                      onLogout={this.logOut}
-                                    />}
-      </div>
-    )
-  }
-}
-
-
-export {AuthenticatedApp};
-export default withRouter(AuthenticatedApp);
+import React, {Component} from 'react';
+import MainScreen from './MainScreen';
+import {Redirect, withRouter} from "react-router-dom";
+import queryString from 'query-string'
+const fetch = require('node-fetch');
+
+class AuthenticatedApp extends Component {
+    constructor() {
+        super();
+        this.state = {
+            username: localStorage.getItem('__username'),
+            access_token: localStorage.getItem('__access_token'),
+            failed_login: false
+        }
+        this.getAuth = this.getAuth.bind(this);
+        this.logOut = this.logOut.bind(this);
+    }
+    componentDidMount() {
+        this._isMounted = true;
+        if (!this.state.access_token) {
+            return this.getAuth();
+        }
+        return fetch('https://api.twitch.tv/helix/users', {
```

Comment:
mostly just curious - what does returning a value from componentDidMount do, anyway? (I feel like I knew this at one point, but if so, it's buried under years of getting rusty)

Reply:
IIRC, if your function uses a promise, it's best to return said promise to avoid any runaway events during unit tests, or something like that